ADR/0047-gitops-onboarding-redesign.md

```diff
@@ -0,0 +1,229 @@
+# 47. GitOps Onboarding Redesign
+
+Date: 2024-12-21
+
+## Status
+
+Proposed
+
+## Context
+
+Currently, Konflux offers two distinct onboarding paths that create friction and complexity for developers:
+
+### UI Onboarding
+- Very easy to use with good validation and feedback
+- Handles secrets, build config, environments through web interface
+- Provides immediate feedback but doesn't align with GitOps principles
+
+### GitOps Onboarding
+- Requires manually authoring YAML across multiple files
+- Validation happens during GitLab CI runs, creating tension between Konflux and GitLab CI workflows
+- Secrets must still be created via UI, breaking the GitOps flow
+- Onboarding is slow and often requires multiple failed PRs to converge due to trial-and-error approach
+
+### Current Pain Points
+1. **Fragmented experience**: Developers must context-switch between UI and GitOps approaches
+2. **Validation delays**: Schema validation only occurs during CI runs, not locally
+3. **Error-prone process**: Manual YAML authoring leads to frequent failures
+4. **Monorepo complexity**: Current GitOps monorepo for tenant configuration creates CODEOWNERS management overhead
+5. **UI improvement limitations**: Despite continuous improvements to the UI, many users still choose GitOps onboarding for its inherent disaster recovery and change management benefits, leaving them unable to benefit from UI enhancements
+
+## Decision
+
+Redesign the onboarding experience to converge UI and GitOps into a single coherent, Git-centric flow through the following architectural changes:
+
+### Alternative Approaches Considered
+
+**Improving the existing UI** was considered but rejected because:
+- Despite continuous UI improvements, many users still choose GitOps onboarding for its inherent disaster recovery and change management benefits
+- UI enhancements don't address the fundamental issue of maintaining two separate onboarding flows
+- Users who prefer GitOps principles cannot benefit from UI improvements, creating a permanent split in the user experience
+
+**Chosen approach** replaces both current flows with a unified Git-centric solution that preserves GitOps benefits while providing the usability improvements users expect.
+
+### Core Shift
+- **Replace UI configuration** with a VS Code plugin for onboarding and editing
+- **Use local schema validation**, linting, and previews to reduce onboarding errors
```

Comment:
This ADR needs a security considerations sections:

* Nothing can ensure that any tenant gitops repo is not full of malicious manifests. Argo configuration needs to be strictly controlled to limit what can be applied to the cluster, and where, from which gitops repo. Brainstorm some specifically bad scenarios in the consequences section - get creative - and discuss how they're mitigated or not.
* Can users see the argo UI? If they can, can they see other tenants' resources? In the monorepo, this is moot because all the resources are there anyways. In the multi-repo model, now it's not clear they should have view rights cross-tenant in the control plane system.
* Can users register a private gitops repo with argo? How does that flow work?